library/src-bootstrapped/scala/quoted/Expr.scala

```diff
@@ -28,11 +28,26 @@ object Expr {
     Block(statements.map(Term.of), Term.of(expr)).asExpr.asInstanceOf[Expr[T]]
   }
 
-  /** Lift a value into an expression containing the construction of that value */
-  def apply[T](x: T)(using lift: Liftable[T])(using Quotes): Expr[T] =
-    lift.toExpr(x)
+  /** Creates an expression that will construct the value `x` */
+  def apply[T](x: T)(using ToExpr[T])(using Quotes): Expr[T] =
+    scala.Predef.summon[ToExpr[T]].apply(x)
 
-  /** Lifts this sequence of expressions into an expression of a sequence
+  /** Get `Some` of a copy of the value if the expression contains a literal constant or constructor of `T`.
+   *  Otherwise returns `None`.
+   *
+   *  Usage:
+   *  ```
+   *  case '{ ... ${expr @ Expr(value)}: T ...} =>
+   *    // expr: Expr[T]
+   *    // value: T
+   *  ```
+   *
+   *  To directly get the value of an expression `expr: Expr[T]` consider using `expr.value`/`expr.valueOrError` insead.
+   */
+  def unapply[T](x: Expr[T])(using FromExpr[T])(using Quotes): Option[T] =
+    scala.Predef.summon[FromExpr[T]].unapply(x)
```

Comment:
I thought about that one. The issue is that it would either fragment concepts into `Expr.apply`/`Value.unapply` or we would need to duplicate some logic between `Expr` and `Value` which might lead to confusions.